src/op_sync_team.rs

```diff
@@ -202,19 +211,24 @@ pub fn sync_team(
             }
         }
         match action {
-            TeamAction::Add(github_name, handle) => {
+            TeamAction::Add(github_name, github_id, handle) => {
                 let logger = logger.new(o!(
                     "nixpkgs-handle" => format!("{}", handle),
                     "github-name" => format!("{}", github_name),
                 ));
-                if pending_invites.contains(&github_name) {
+                if pending_invites.contains(&github_name) || invited.contains(&github_id) {
                     noops.inc();
-                    debug!(logger, "User already has a pending invitation");
+                    debug!(logger, "User has already been invited previously and/or still has a pending invitation");
                 } else {
                     additions.inc();
                     info!(logger, "Adding user to the team");
 
                     if do_it_live {
+                        // keep track of the invitation locally so that we don't
+                        // spam users that have already been invited and rejected
+                        // the invitation
+                        invited.add(github_id.clone());
```

Comment:
Same as above (users should only be added if they were successfully invited). There's already a `match` statement ready for you to add on to, for this one.